components/gfx/paint_context.rs

```diff
@@ -83,6 +91,23 @@ struct Line {
     start: Point2D<f32>,
```

Comment:
Add a comment to this struct stating the invariant that you're now upholding in the constructor